dashboard/app/assets/stylesheets/DanceAiModal.scss

```diff
@@ -0,0 +1,105 @@
+// Fade in for 1 second after a 0 second delay.
+@keyframes appear0 {
+  0% {
+    opacity: 0;
+  }
+  100% {
+    opacity: 1;
+  }
+}
+
+// Fade in for 1 second after a 1 second delay.
+@keyframes appear1 {
+  0%, 50% {
+    opacity: 0;
+  }
+  100% {
+    opacity: 1;
+  }
+}
+
+// Fade in for 1 second after a 2 second delay.
+@keyframes appear2 {
+  0%, 66% {
+    opacity: 0;
+  }
+  100% {
+    opacity: 1;
+  }
+}
+
+// Fade in for 1 second after a 3 second delay.
+@keyframes appear3 {
+  0%, 75% {
+    opacity: 0;
+  }
+  100% {
+    opacity: 1;
+  }
+}
+
+// Fade in for 1 second after a 4 second delay.
+@keyframes appear4 {
+  0%, 80% {
+    opacity: 0;
+  }
+  100% {
+    opacity: 1;
+  }
+}
+
+// Fade in for 1 second after a 5 second delay.
+@keyframes appear5 {
+  0%, 83% {
+    opacity: 0;
+  }
+  100% {
+    opacity: 1;
+  }
+}
+
+// Fade in for 1 second after a 6 second delay.
+@keyframes appear6 {
+  0%, 85% {
+    opacity: 0;
+  }
+  100% {
+    opacity: 1;
+  }
+}
+
+#ai-modal-inner-area {
+  #fade-in {
+    // First block
+    .blocklyWorkspace .blocklyPath {
```

Comment:
Looks like these class names are very specific to the blockly workspace layout of the specific blocks we're previewing. Maybe worth dropping a comment somewhere to indicate that if the block preview changes at all, then these style rules need to be updated too?

Reply:
Good call.  Added a big comment to this file.